¿Si usted fuera la entrenadora, como ordena el valor esperado de encestes en una ronda de 100 tiros libres a cada una de las candidatas propuestas por las ocho cazatalentos ? donde Ci < Cj sii prob( #Encestes Ci < #Encestes Cj | 100 tiros cada uno ) > 0.5

es decir que, haciendo 100 tiros libres, más de la mitad de las veces, Ci va a lograr menos
encestes que Cj

**Cazatalentos 1**: Llegué a la localidad habían 100 adolescentes en el gimnasio, hice tirar a cada una
100 tiros, sólo una obtuvo 80 aciertos, luego un par 79, otro par 78, y así descendían.

Resolución: Este es análogo al "Caso 2: Aparece el Overfitting". Con 100 tiros, se puede suponer que la basquetbolista tuvo una esperanza de encestes de 0.8, y luego una sucesión de las dos siguientes 79 , luego 78, y así. 

Nota: Que haya sido una sóla ronda no garatiza que no haya sido la "malidción del ganador", donde si se repite, la basquetbolista ganadora (sobre todo al haber tan poca diferencia entre los primeros lugares) confirme luego su performance

Por ello, luego habría que simular una nueva ronda para comparar

In [1]:
import numpy as np

#Pongo mi semilla[1]
np.random.seed(279511)

In [2]:
# calcula cuantos encestes logra un jugadora con indice de enceste prob
# haciendo qyt tiros libres

def ftirar(prob, qty):
  return sum(np.random.rand(qty) < prob)


# defino los jugadoras
mejor = 0.8

In [3]:
#Defino el pelotón de jugadores siguientes
inicio = 79
final = 30
paso = -1
repeticiones = 2
total_elementos = 99 - 70 + 1

# Creo el array de secuencia descendente
secuencia = np.arange(inicio, final + paso, paso)

# Repito la secuencia 2 veces
secuencia_repetida = np.repeat(secuencia, repeticiones)

# Dividir cada elemento por 100
peloton = secuencia_repetida / 100

print(peloton)

[0.79 0.79 0.78 0.78 0.77 0.77 0.76 0.76 0.75 0.75 0.74 0.74 0.73 0.73
 0.72 0.72 0.71 0.71 0.7  0.7  0.69 0.69 0.68 0.68 0.67 0.67 0.66 0.66
 0.65 0.65 0.64 0.64 0.63 0.63 0.62 0.62 0.61 0.61 0.6  0.6  0.59 0.59
 0.58 0.58 0.57 0.57 0.56 0.56 0.55 0.55 0.54 0.54 0.53 0.53 0.52 0.52
 0.51 0.51 0.5  0.5  0.49 0.49 0.48 0.48 0.47 0.47 0.46 0.46 0.45 0.45
 0.44 0.44 0.43 0.43 0.42 0.42 0.41 0.41 0.4  0.4  0.39 0.39 0.38 0.38
 0.37 0.37 0.36 0.36 0.35 0.35 0.34 0.34 0.33 0.33 0.32 0.32 0.31 0.31
 0.3  0.3 ]


In [4]:
# Le saco el último elemento del array
peloton = peloton[:-1]
len(peloton)

99

In [5]:
jugadoras = np.append(mejor, peloton)
# Reviso jugadoras
jugadoras

array([0.8 , 0.79, 0.79, 0.78, 0.78, 0.77, 0.77, 0.76, 0.76, 0.75, 0.75,
       0.74, 0.74, 0.73, 0.73, 0.72, 0.72, 0.71, 0.71, 0.7 , 0.7 , 0.69,
       0.69, 0.68, 0.68, 0.67, 0.67, 0.66, 0.66, 0.65, 0.65, 0.64, 0.64,
       0.63, 0.63, 0.62, 0.62, 0.61, 0.61, 0.6 , 0.6 , 0.59, 0.59, 0.58,
       0.58, 0.57, 0.57, 0.56, 0.56, 0.55, 0.55, 0.54, 0.54, 0.53, 0.53,
       0.52, 0.52, 0.51, 0.51, 0.5 , 0.5 , 0.49, 0.49, 0.48, 0.48, 0.47,
       0.47, 0.46, 0.46, 0.45, 0.45, 0.44, 0.44, 0.43, 0.43, 0.42, 0.42,
       0.41, 0.41, 0.4 , 0.4 , 0.39, 0.39, 0.38, 0.38, 0.37, 0.37, 0.36,
       0.36, 0.35, 0.35, 0.34, 0.34, 0.33, 0.33, 0.32, 0.32, 0.31, 0.31,
       0.3 ])

In [6]:
# Me fijo qué sucede si le hago realizar 100 tiros libres a las 100 jugadoras

# vectorizo la funcion  ftirar
vec_ftirar = np.vectorize(ftirar)

# hago que las 100 jugadoras tiren 100 veces cada una
vec_ftirar(jugadoras, 100)

primera_ganadora = 0

for i in range(10000): # diez mil experimentos
  vaciertos = vec_ftirar(jugadoras, 100) # 100 tiros libres cada jugadora
  mejor_ronda = np.argmax(vaciertos)
  if mejor_ronda == 0:
    primera_ganadora += 1




print(primera_ganadora)

2312


En 10.000 experimentos, sólo en 2312 casos la mejor jugadora que eligió la cazatalento 1 fue la mejor...

- **Cazatalentos 2**: Llegué a una pequeña ciudad y habían 200 adolescentes en el gimnasio, hice tirar
a cada una 100 tiros, sólo una obtuvo 80 aciertos, seis 79, cinco 78, y así descendían.

Caso análogo al primero

In [7]:
# defino los jugadoras
mejor = 0.8

In [8]:
# Crear una secuencia descendente para el pelotón inicial
secuencia1 = [np.repeat(80 - x, 7 - x) for x in range(1, 7)]
peloton1 = np.concatenate(secuencia1)

# Crear una secuencia descendente para el segundo pelotón
peloton2 = np.arange(73, 9, -1)

# Combinar ambas secuencias
peloton = np.concatenate((peloton1, peloton2))

# Asegurarse de que la longitud sea de 199 elementos
while len(peloton) < 199:
    peloton = np.append(peloton, 10)

# Dividir todos los elementos por 100
peloton = peloton / 100

print(peloton)

[0.79 0.79 0.79 0.79 0.79 0.79 0.78 0.78 0.78 0.78 0.78 0.77 0.77 0.77
 0.77 0.76 0.76 0.76 0.75 0.75 0.74 0.73 0.72 0.71 0.7  0.69 0.68 0.67
 0.66 0.65 0.64 0.63 0.62 0.61 0.6  0.59 0.58 0.57 0.56 0.55 0.54 0.53
 0.52 0.51 0.5  0.49 0.48 0.47 0.46 0.45 0.44 0.43 0.42 0.41 0.4  0.39
 0.38 0.37 0.36 0.35 0.34 0.33 0.32 0.31 0.3  0.29 0.28 0.27 0.26 0.25
 0.24 0.23 0.22 0.21 0.2  0.19 0.18 0.17 0.16 0.15 0.14 0.13 0.12 0.11
 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1
 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1
 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1
 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1
 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1
 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1
 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1
 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1
 0.1  0.1  0.1

In [9]:
len(peloton)

199

In [10]:
jugadoras = np.append(mejor, peloton)
# Reviso jugadoras
jugadoras

array([0.8 , 0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.78, 0.78, 0.78, 0.78,
       0.78, 0.77, 0.77, 0.77, 0.77, 0.76, 0.76, 0.76, 0.75, 0.75, 0.74,
       0.73, 0.72, 0.71, 0.7 , 0.69, 0.68, 0.67, 0.66, 0.65, 0.64, 0.63,
       0.62, 0.61, 0.6 , 0.59, 0.58, 0.57, 0.56, 0.55, 0.54, 0.53, 0.52,
       0.51, 0.5 , 0.49, 0.48, 0.47, 0.46, 0.45, 0.44, 0.43, 0.42, 0.41,
       0.4 , 0.39, 0.38, 0.37, 0.36, 0.35, 0.34, 0.33, 0.32, 0.31, 0.3 ,
       0.29, 0.28, 0.27, 0.26, 0.25, 0.24, 0.23, 0.22, 0.21, 0.2 , 0.19,
       0.18, 0.17, 0.16, 0.15, 0.14, 0.13, 0.12, 0.11, 0.1 , 0.1 , 0.1 ,
       0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 ,
       0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 ,
       0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 ,
       0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 ,
       0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 ,
       0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.1 , 0.

In [11]:
# Me fijo qué sucede si le hago realizar 100 tiros libres a las 200 jugadoras

# vectorizo la funcion  ftirar
vec_ftirar = np.vectorize(ftirar)

# hago que las 100 jugadoras tiren 200 veces cada una
vec_ftirar(jugadoras, 200)

primera_ganadora = 0

for i in range(10000): # diez mil experimentos
  vaciertos = vec_ftirar(jugadoras, 200) # 200 tiros libres cada jugadora
  mejor_ronda = np.argmax(vaciertos)
  if mejor_ronda == 0:
    primera_ganadora += 1




print(primera_ganadora)

1650


En 10.000 experimentos, sólo en 1650 casos la mejor jugadora que eligió la cazatalento 21 fue la mejor...

- **Cazatalentos 3**: Llegué a un minúsculo pueblo con un aro reglamentario en la pared de una
escuela, había solamente 2 adolescentes, hice tirar a cada una 100 tiros, una obtuvo 80 y la otra 75

Caso similar, pero menos participantes

In [12]:
# defino los jugadoras
mejor = 0.8
peloton = 0.7
jugadoras = np.append(mejor, peloton)

In [13]:
# Me fijo qué sucede si le hago realizar 100 tiros libres a las 2 jugadoras

# vectorizo la funcion  ftirar
vec_ftirar = np.vectorize(ftirar)

# hago que las 2 jugadoras tiren 100 veces cada una
vec_ftirar(jugadoras, 100)

primera_ganadora = 0

for i in range(10000): # diez mil experimentos
  vaciertos = vec_ftirar(jugadoras, 100) # 100 tiros libres cada jugadora
  mejor_ronda = np.argmax(vaciertos)
  if mejor_ronda == 0:
    primera_ganadora += 1




print(primera_ganadora)

9566


Más seguro que esta jugadora ha sido la mejor de estas 2: en 9566 de la 10.000 veces ganó, aunque muy baja la comparación

- **Cazatalentos 4**: Ehh, yo soy nueva y no tenía clara la regla de hacer tirar 100 tiros. Fui a un pueblo,
había 100 adolescentes, le hice tirar 10 tiros a cada una. La mejor de ellas acierta 9 y la elegí como ganadora, luego ya hay varias con 8 aciertos, 7, etc.

Similar al primer caso. Lo interpreto como que va descendiendo entre 0.8 a 0.7 con 100 elementos en el pelotón

In [14]:
# defino los jugadoras
mejor = 0.9

In [19]:
peloton = np.array(range(800, 701, -1)) / 1000
peloton

array([0.8  , 0.799, 0.798, 0.797, 0.796, 0.795, 0.794, 0.793, 0.792,
       0.791, 0.79 , 0.789, 0.788, 0.787, 0.786, 0.785, 0.784, 0.783,
       0.782, 0.781, 0.78 , 0.779, 0.778, 0.777, 0.776, 0.775, 0.774,
       0.773, 0.772, 0.771, 0.77 , 0.769, 0.768, 0.767, 0.766, 0.765,
       0.764, 0.763, 0.762, 0.761, 0.76 , 0.759, 0.758, 0.757, 0.756,
       0.755, 0.754, 0.753, 0.752, 0.751, 0.75 , 0.749, 0.748, 0.747,
       0.746, 0.745, 0.744, 0.743, 0.742, 0.741, 0.74 , 0.739, 0.738,
       0.737, 0.736, 0.735, 0.734, 0.733, 0.732, 0.731, 0.73 , 0.729,
       0.728, 0.727, 0.726, 0.725, 0.724, 0.723, 0.722, 0.721, 0.72 ,
       0.719, 0.718, 0.717, 0.716, 0.715, 0.714, 0.713, 0.712, 0.711,
       0.71 , 0.709, 0.708, 0.707, 0.706, 0.705, 0.704, 0.703, 0.702])

In [20]:
len(peloton)

99

In [21]:
jugadoras = np.append(mejor, peloton)
jugadoras

array([0.9  , 0.8  , 0.799, 0.798, 0.797, 0.796, 0.795, 0.794, 0.793,
       0.792, 0.791, 0.79 , 0.789, 0.788, 0.787, 0.786, 0.785, 0.784,
       0.783, 0.782, 0.781, 0.78 , 0.779, 0.778, 0.777, 0.776, 0.775,
       0.774, 0.773, 0.772, 0.771, 0.77 , 0.769, 0.768, 0.767, 0.766,
       0.765, 0.764, 0.763, 0.762, 0.761, 0.76 , 0.759, 0.758, 0.757,
       0.756, 0.755, 0.754, 0.753, 0.752, 0.751, 0.75 , 0.749, 0.748,
       0.747, 0.746, 0.745, 0.744, 0.743, 0.742, 0.741, 0.74 , 0.739,
       0.738, 0.737, 0.736, 0.735, 0.734, 0.733, 0.732, 0.731, 0.73 ,
       0.729, 0.728, 0.727, 0.726, 0.725, 0.724, 0.723, 0.722, 0.721,
       0.72 , 0.719, 0.718, 0.717, 0.716, 0.715, 0.714, 0.713, 0.712,
       0.711, 0.71 , 0.709, 0.708, 0.707, 0.706, 0.705, 0.704, 0.703,
       0.702])

In [22]:
# vectorizo la funcion  ftirar
vec_ftirar = np.vectorize(ftirar)

# hago que las 100 jugadoras tiren 10 veces cada una
vec_ftirar(jugadoras, 10)

primera_ganadora = 0

for i in range(10000): # diez mil experimentos
  vaciertos = vec_ftirar(jugadoras, 10) # 10 tiros libres cada jugadora
  mejor_ronda = np.argmax(vaciertos)
  if mejor_ronda == 0:
    primera_ganadora += 1




print(primera_ganadora)

3652


En 10.000 experimentos, sólo en 3652 casos la mejor jugadora que eligió la cazatalento 21 fue la mejor...

- **Cazatalentos 5**: Había 100 adolescentes, las hice tirar una primer ronda de 100 tiros, me quedé con las 5 mejores de esa primer ronda, y luego hice dos rondas más de 100 tiros, y los resultados fueron los siguentes:

![image.png](attachment:image.png)

Elijo jugadora A

Acá primero es como el caso del teórico "Tercer Caso" (nueva ronda) y "Cuarto Caso" (ganador de nueva ronda, overfitting en su máxima expresión), para explicar la "Maldición del Ganador"